test/pkg/test/framework.go

```diff
@@ -102,13 +101,12 @@ func (env *Framework) ExpectToBeClean(ctx context.Context) {
 		env.EventuallyExpectNoneFound(ctx, testObject.ListType)
 	})
 
-	currentClusterVpcId := os.Getenv("CLUSTER_VPC_ID")
-	retrievedServiceNetworkVpcAssociations, _ := env.LatticeClient.ListServiceNetworkVpcAssociationsAsList(ctx, &vpclattice.ListServiceNetworkVpcAssociationsInput{
-		VpcIdentifier: aws.String(currentClusterVpcId),
-	})
-	Logger(ctx).Infof("Expect VPC used by current cluster don't have any ServiceNetworkVPCAssociation, if it has you should manually delete it")
-	Expect(len(retrievedServiceNetworkVpcAssociations)).To(Equal(0))
 	Eventually(func(g Gomega) {
+		retrievedServiceNetworkVpcAssociations, _ := env.LatticeClient.ListServiceNetworkVpcAssociationsAsList(ctx, &vpclattice.ListServiceNetworkVpcAssociationsInput{
+			VpcIdentifier: aws.String(CurrentClusterVpcId),
+		})
+		Logger(ctx).Infof("Expect VPC used by current cluster don't have any ServiceNetworkVPCAssociation, if it has you should manually delete it")
+		g.Expect(len(retrievedServiceNetworkVpcAssociations)).To(Equal(0))
```

Comment:
Not sure if it's improvement, I dont see any good of making Eventually assertions even more bigger. We should get away from these gigantic Eventually blocks, it does not looks idiomatic from ginkgo framework. I see that we use eventually as a big retry on error block, that's not exact purpose of Eventually